src/main/java/org/fenixedu/academic/domain/studentCurriculum/StudentCurricularPlanEnrolmentInExtraordinarySeasonEvaluationManager.java

```diff
@@ -0,0 +1,197 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FenixEdu Academic is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with FenixEdu Academic.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.fenixedu.academic.domain.studentCurriculum;
+
+import org.fenixedu.academic.domain.Enrolment;
+import org.fenixedu.academic.domain.EnrolmentEvaluation;
+import org.fenixedu.academic.domain.ExecutionYear;
+import org.fenixedu.academic.domain.accounting.EnrolmentBlocker;
+import org.fenixedu.academic.domain.accounting.events.EnrolmentEvaluationEvent;
+import org.fenixedu.academic.domain.curricularRules.ICurricularRule;
+import org.fenixedu.academic.domain.curricularRules.executors.ruleExecutors.CurricularRuleLevel;
+import org.fenixedu.academic.domain.curricularRules.executors.ruleExecutors.EnrolmentResultType;
+import org.fenixedu.academic.domain.enrolment.EnroledCurriculumModuleWrapper;
+import org.fenixedu.academic.domain.enrolment.EnrolmentContext;
+import org.fenixedu.academic.domain.enrolment.IDegreeModuleToEvaluate;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.domain.student.Registration;
+import org.fenixedu.academic.domain.student.Student;
+import org.fenixedu.academic.domain.student.registrationStates.RegistrationStateType;
+
+import java.util.*;
+import java.util.Map.Entry;
+
+public class StudentCurricularPlanEnrolmentInExtraordinarySeasonEvaluationManager extends StudentCurricularPlanEnrolment {
+
+    public StudentCurricularPlanEnrolmentInExtraordinarySeasonEvaluationManager(final EnrolmentContext enrolmentContext) {
+        super(enrolmentContext);
+    }
+
+    @Override
+    protected void assertEnrolmentPreConditions() {
+        if (!getRegistration().hasStateType(getExecutionYear(), RegistrationStateType.REGISTERED)) {
+            throw new DomainException("error.StudentCurricularPlan.cannot.enrol.with.registration.inactive");
+        }
+
+        super.assertEnrolmentPreConditions();
+    }
+
+    @Override
+    protected void checkDebts() {
+    }
+
+    @Override
+    protected void assertAcademicAdminOfficePreConditions() {
+
+        checkEnrolmentWithoutRules();
+
+        if (updateRegistrationAfterConclusionProcessPermissionEvaluated()) {
+            return;
+        }
+    }
+
+    @Override
+    protected void assertStudentEnrolmentPreConditions() {
+
+        if (!getRegistrationsToEnrolByStudent(getResponsiblePerson().getStudent()).contains(getRegistration())) {
+            throw new DomainException("error.StudentCurricularPlan.student.is.not.allowed.to.perform.enrol");
+        }
+
```

Comment:
A change in requirements… Similarly to the special season case, we need to explicitly check StatuteType to allow each student access to this evaluation season.

Here we neet to add that check. This means we will also need an extra flag in the StatuteType: extraordinarySeasonGranted